RELEASE.md

```diff
@@ -0,0 +1,304 @@
+# Release process
+
+The release process starts with one or more tagged release candidates,
+when no more blocking issues needs to be fixed a final tag is created
+and the final release is rolled.
+
+confluent-kafka-dotnet uses semver versioning and loosely follows
+librdkafka's version, e.g. v0.11.x for the final release and
+v0.11.x-RC3 for the 3rd v0.11.x release candidate.
+
+With the addition of prebuilt binary packages we make use of travis-ci.org.
+These artifacts are downloaded from artifacts url of each ci build (for 
+example: https://ci.appveyor.com/project/ConfluentClientEngineering/confluent-kafka-dotnet/builds/39236976/artifacts),
+and then uploaded manually to https://www.nuget.org/.
+
+**Note**: Dotnet package versions use the same tag with librdkafka.
+          That is to say that while the librdkafka RC is named `v0.11.x-RC3`,
+          a dotnet client RC with the same version is also named `v0.11.x-RC3`.
+
+
+The following guide uses `v0.11.x-RC1` as version for a release candidate,
```

Comment:
does it? seems a bit all over the place, (e.g. 1.7.x mentioned), I would remove this paragraph as well as unnecessary.